pep-0688.rst

```diff
@@ -0,0 +1,275 @@
+PEP: 688
+Title: Making the buffer protocol accessible in Python
+Author: Jelle Zijlstra <jelle.zijlstra@gmail.com>
+Status: Draft
+Type: Standards Track
+Content-Type: text/x-rst
+Created: 23-Apr-2022
+Python-Version: 3.12
+
+
+Abstract
+========
+
+This PEP proposes a mechanism for Python code to inspect whether a
+type supports the C-level buffer protocol. This allows type
+checkers to evaluate whether objects implement the protocol.
+
+
+Motivation
+==========
+
+The CPython C API provides a versatile mechanism for accessing the
+underlying memory of an object—the buffer protocol from :pep:`3118`.
+Functions that accept binary data are usually written to handle any
+object implementing the buffer protocol. For example, at the time of writing,
+there are about 130 functions in CPython using the Argument Clinic
+``Py_buffer`` type, which accepts the buffer protocol.
+
+Currently, there is no way for Python code to inspect whether an object
+supports the buffer protocol. Moreover, the static type system
+does not provide a type annotation to represent the protocol.
+This is a `common problem <https://github.com/python/typing/issues/593>`__
+when type-annotating code that accepts generic buffers.
```

Comment:
```suggestion
when adding type annotations to code that accepts generic buffers.
```

Reply:
Going to go with "writing type annotations for code", which I think better describes typeshed